packages/wallet/src/redux/protocols/concluding/reducer.ts

```diff
@@ -1,44 +1,238 @@
+  protocolState: states.WaitForDefund,
   sharedData: SharedData,
-) {
-  const result = initializeInstigator(channelId, processId, sharedData);
-  return { protocolState: result.protocolState, sharedData: result.sharedData };
+  action: ProtocolAction,
+): ProtocolStateWithSharedData<ConcludingState> {
+  if (!routesToDefunding(action, EMPTY_LOCATOR)) {
+    console.warn(`Expected defunding, received ${action.type} instead`);
+    return { protocolState, sharedData };
+  }
+
+  let defunding: DefundingState;
+  ({ protocolState: defunding, sharedData } = defundingReducer(
+    protocolState.defunding,
+    sharedData,
+    action,
+  ));
+
+  switch (defunding.type) {
+    case 'Defunding.Failure':
+      sharedData = sendConcludeFailure(sharedData, 'Other');
+      return { protocolState: states.failure({ reason: 'Defunding failure' }), sharedData };
+    case 'Defunding.Success':
+      return { protocolState: states.decideClosing(protocolState), sharedData };
+
+    default:
+      return {
+        protocolState: states.waitForDefund({ ...protocolState, defunding }),
+        sharedData,
+      };
+  }
 }
 
-export function initializeResponderState(
-  signedCommitment: SignedCommitment,
-  processId: string,
+function waitForConcludeReducer(
+  protocolState: states.WaitForConclude,
   sharedData: SharedData,
-) {
-  const result = initializeResponder(signedCommitment, processId, sharedData);
-  return { protocolState: result.protocolState, sharedData: result.sharedData };
+  action: ProtocolAction,
+): ProtocolStateWithSharedData<ConcludingState> {
+  if (!routesToAdvanceChannel(action, EMPTY_LOCATOR)) {
+    console.warn(`Expected Advance channel action, received ${action.type} instead`);
+    return { protocolState, sharedData };
+  }
+  let concluding: AdvanceChannelState;
+  ({ protocolState: concluding, sharedData } = advanceChannelReducer(
+    protocolState.concluding,
+    sharedData,
+    action,
+  ));
+  switch (concluding.type) {
+    case 'AdvanceChannel.Failure':
+      sharedData = sendConcludeFailure(sharedData, 'Other');
+      return { protocolState: states.failure({ reason: 'AdvanceChannelAction' }), sharedData };
+    case 'AdvanceChannel.Success':
+      let defunding: DefundingState;
+      const { processId, channelId } = protocolState;
+      ({ protocolState: defunding, sharedData } = initializeDefunding(
+        processId,
+        makeLocator(EmbeddedProtocol.Defunding),
+        channelId,
+        sharedData,
+      ));
+      return {
+        protocolState: states.waitForDefund({ ...protocolState, defunding }),
+        sharedData,
+      };
+    default:
+      return {
+        protocolState: states.waitForConclude({ ...protocolState, concluding }),
+        sharedData,
+      };
+  }
+}
+
+export function initialize({
+  channelId,
+  processId,
+  opponentInstigatedConclude,
+  sharedData,
+}: {
+  channelId: string;
+  processId: string;
+  opponentInstigatedConclude: boolean;
```

Comment:
Does it make more sense for this to be `weInstigatedConclude`?

Reply:
I think either make sense to me, is there a reason you prefer `weInstigatedConclude` ?